src/loop_logic.rs

```diff
@@ -892,6 +990,62 @@ mod tests {
         drop(sender);
     }
 
+    #[cfg(feature = "block_on")]
+    #[test]
+    fn block_on_test() {
+        use std::time::Duration;
+
+        let mut data = 22;
+        let timeout = {
+            let data = &mut data;
+
+            async move {
+                async_io::Timer::after(Duration::from_secs(1)).await;
+                *data = 32;
+                11
+            }
```

Comment:
calloop has its own timer futures (`calloop::timer::TimeoutFuture`), we can use them instead of pulling async-io as a dev dependency. Furthermore this would ensure that the added function is indeed able to work with calloop-driven futures.